app/actions/navigation.js

```diff
@@ -108,6 +152,119 @@ export function goToScreen(componentId, name, title, passProps = {}, options = {
             },
         };
 
+        Navigation.push(componentId, {
+            component: {
+                name,
+                passProps,
+                options: merge(defaultOptions, options),
+            },
+        });
+    };
+}
+
+export function showModal(name, title, passProps = {}, options = {}) {
+    return (dispatch, getState) => {
+        const theme = getTheme(getState());
+        const defaultOptions = {
+            layout: {
+                backgroundColor: theme.centerChannelBg,
+            },
+            statusBar: {
+                visible: true,
+            },
+            topBar: {
+                animate: true,
+                visible: true,
+                backButton: {
+                    color: theme.sidebarHeaderTextColor,
+                    title: '',
+                },
+                background: {
+                    color: theme.sidebarHeaderBg,
+                },
+                title: {
+                    color: theme.sidebarHeaderTextColor,
+                    text: title,
+                },
+            },
+        };
+
+        Navigation.showModal({
+            stack: {
+                children: [{
+                    component: {
+                        name,
+                        passProps,
+                        options: merge(defaultOptions, options),
+                    },
+                }],
+            },
+        });
+    };
+}
+
+export function showModalOverCurrentContext(name, passProps = {}, options = {}) {
+    return (dispatch) => {
+        const title = '';
+        const animationsEnabled = (Platform.OS === 'android').toString();
+        const defaultOptions = {
+            modalPresentationStyle: 'overCurrentContext',
+            layout: {
+                backgroundColor: 'transparent',
+            },
+            topBar: {
+                visible: false,
+                height: 0,
+            },
+            animations: {
+                showModal: {
+                    enabled: animationsEnabled,
+                    alpha: {
+                        from: 0,
+                        to: 1,
+                        duration: 250,
+                    },
+                },
+                dismissModal: {
+                    enabled: animationsEnabled,
+                    alpha: {
+                        from: 1,
+                        to: 0,
+                        duration: 250,
+                    },
+                },
+            },
+        };
+        const mergeOptions = merge(defaultOptions, options);
+
+        dispatch(showModal(name, title, passProps, mergeOptions));
+    };
+}
+
+export function showSearchModal(initialValue = '') {
+    return (dispatch) => {
```

Comment:
Minor nit, but this can just be 
```javascript
export function showSearchModal(initialValue = '') {
    // ...

    showModal(name, title, passProps, options));
}
```
without the intermediate function